src/main/scala/smile/modeling/Transformer.scala

```diff
@@ -0,0 +1,590 @@
+package smile.modeling
+
+import smile.infrastructure.MathUtils
+
+/** Contains methods for transforming positions, particularly for rotation and scaling operations.
+  */
+object Transformer:
```

Comment:
Is it redundant that there's `AffineTransformation` and `Transformer`? Difference?